### PR TITLE
feat: perform license check on startup - re-add functionality

### PR DESCRIPTION
**Pull request by matthewBearCamunda**
## Description

This PR readds the code from https://github.com/camunda/camunda/pull/20541 which was reverted in https://github.com/camunda/camunda/pull/20940

This PR can be merged once the license checker library has been moved to a public repo
- Completed with https://github.com/camunda/camunda-license-check/pull/34
- Update license checker version to 2.9.0, the version which was uploaded to `camunda-bpm`

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

**Comment by matthewBearCamunda** [comment]
I was able to download the new `2.9.0` version of `camunda-license-checker` locally during a build. I confirmed with another member of the team as well, they were able to download the dependency during a build

File: parent/pom.xml [hunk 1]
```diff
@@ -40,6 +40,7 @@
     <version.awaitility>4.2.2</version.awaitility>
     <version.bouncycastle>1.78.1</version.bouncycastle>
     <version.camunda>7.22.0-alpha4</version.camunda>
+    <version.camunda-license-check>2.9.0</version.camunda-license-check>
```
Comment (Zelldon) [inline parent/pom.xml:43]: ❓ I assume that this is the recent version that has been published/released now on a open maven repo?

**Review by ChrisKujawa — APPROVED** [review]
Great thanks @matthewBearCamunda for handling this 👍🏼

File: parent/pom.xml (hunk 1 above)
Comment (matthewBearCamunda) [reply parent/pom.xml:43]: Yes! It was moved with https://github.com/camunda/camunda-license-check/pull/34

File: bom/pom.xml [hunk 2]
```diff
@@ -152,6 +152,30 @@
...
     </repository>
 
+    <repository>
+      <releases>
+        <enabled>true</enabled>
+      </releases>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+      <id>camunda-cpm</id>
```
Comment (yanavasileva) [inline bom/pom.xml:162]: @matthewBearCamunda,
🙃 nitpick: there's a typo here.

**Pull request merged**